### PR TITLE
Fix RecordDownloaderTest

### PR DESCRIPTION
**Pull request by steven-sheehy**
**Detailed description**:
- Fix issue with tests not being detected because they were renamed to `TestV1` and `TestV2`
- Consolidates v1 and v2 tests into a single test
- Switches majority of tests to use v2 data from testnet since that is the type we will use in mainnet
- Single v1 test added just to do a quick backwards compatibility test
- Restructed folders from `recordstreamsV1` to `recordstreams/v1`

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Checklist**
- [ ] Documentation added
- [x] Tests updated

### CODE REVIEW

File: src/test/java/com/hedera/downloader/RecordFileDownloaderTest.java [hunk 1]
```diff
@@ -83,17 +83,35 @@ void after() {
...
 
+    @Test
+    @DisplayName("Download and verify V2 files")
+    void downloadV2() throws Exception {
+        fileCopier.copy();
+        downloader.download();
+        verify(applicationStatus).updateLastValidDownloadedRecordFileName("2019-08-30T18_10_05.249678Z.rcd");
+        assertThat(Files.walk(validPath))
+                .filteredOn(p -> !p.toFile().isDirectory())
+                .hasSize(2)
```
Comment (mike-burrage-hedera) [inline src/test/java/com/hedera/downloader/RecordFileDownloaderTest.java]: Could we actually validate the 2 filenames here. Just to make sure the 2 actual correct rcd files (by name) are in the valid directory. I have the same comment for the rest of the changes where we validate the number of files in the directory. I'd prefer actually validating the filename(s).

File: src/test/java/com/hedera/downloader/RecordFileDownloaderTest.java [hunk 2]
```diff
@@ -164,11 +182,11 @@ void hashMismatchWithPrevious() throws Exception {
         when(applicationStatus.getLastValidDownloadedRecordFileName()).thenReturn("2019-07-01T14:12:00.000000Z.rcd");
         when(applicationStatus.getLastValidDownloadedRecordFileHash()).thenReturn("123");
         when(applicationStatus.getBypassRecordHashMismatchUntilAfter()).thenReturn("");
-        fileCopier.copy();
+        fileCopier.filterFiles("2019-08-30T18_10_05.249678Z.rcd*").copy(); // Skip first file with zero hash
         downloader.download();
         assertThat(Files.walk(validPath))
                 .filteredOn(p -> !p.toFile().isDirectory())
-                .hasSize(2)
+                .hasSize(1)
```
Comment (mike-burrage-hedera) [inline src/test/java/com/hedera/downloader/RecordFileDownloaderTest.java]: Would prefer validating the file name.

File: src/test/java/com/hedera/downloader/RecordFileDownloaderTest.java [hunk 3]
```diff
@@ -178,19 +196,19 @@ void hashMismatchWithBypass() throws Exception {
         when(applicationStatus.getLastValidDownloadedRecordFileName()).thenReturn("2019-07-01T14:12:00.000000Z.rcd");
         when(applicationStatus.getLastValidDownloadedRecordFileHash()).thenReturn("123");
         when(applicationStatus.getBypassRecordHashMismatchUntilAfter()).thenReturn("2019-07-02T00:00:00.000000Z.rcd");
-        fileCopier.copy();
+        fileCopier.filterFiles("2019-08-30T18_10_05.249678Z.rcd*").copy(); // Skip first file with zero hash
         downloader.download();
         assertThat(Files.walk(validPath))
                 .filteredOn(p -> !p.toFile().isDirectory())
-                .hasSize(2)
+                .hasSize(1)
```
Comment (mike-burrage-hedera) [inline src/test/java/com/hedera/downloader/RecordFileDownloaderTest.java]: Would prefer validating the file name.

File: src/test/java/com/hedera/downloader/RecordFileDownloaderTest.java (hunk 1 above)
Comment (steven-sheehy) [reply src/test/java/com/hedera/downloader/RecordFileDownloaderTest.java]: Done. I hardcoded filenames for every test that has a non-zero number of files.

File: src/test/java/com/hedera/downloader/RecordFileDownloaderTest.java (hunk 2 above)
Comment (steven-sheehy) [reply src/test/java/com/hedera/downloader/RecordFileDownloaderTest.java]: Done

File: src/test/java/com/hedera/downloader/RecordFileDownloaderTest.java (hunk 3 above)
Comment (steven-sheehy) [reply src/test/java/com/hedera/downloader/RecordFileDownloaderTest.java]: Done

**Pull request merged**